router/src/main/kotlin/io/moia/router/RequestHandler.kt

```diff
@@ -126,73 +146,71 @@ abstract class RequestHandler : RequestHandler<APIGatewayProxyRequestEvent, APIG
-                        UnprocessableEntityError(
+                        )
+                    )
+                )
+            )
+            is InvalidFormatException -> ResponseEntity(
+                422,
+                createUnprocessableEntityErrorBody(
+                    UnprocessableEntityError(
                         message = "INVALID_FIELD_FORMAT",
                         code = "FIELD",
-                        path = ex.path.last().fieldName.orEmpty()))))
-            is MissingKotlinParameterException ->
-                createResponse(contentType, input,
-                    ResponseEntity(422, createUnprocessableEntityErrorBody(UnprocessableEntityError(
+                        path = ex.path.last().fieldName.orEmpty()
+                    )
+                )
+            )
+            is MissingKotlinParameterException -> ResponseEntity(
+                422,
+                createUnprocessableEntityErrorBody(
+                    UnprocessableEntityError(
                         message = "MISSING_REQUIRED_FIELDS",
                         code = "FIELD",
-                        path = ex.parameter.name.orEmpty()))))
-            else -> createResponse(contentType, input,
-                ResponseEntity(500, createErrorBody(ApiError(ex.message.orEmpty(), "INTERNAL_SERVER_ERROR"))))
+                        path = ex.parameter.name.orEmpty()
+                    )
+                )
+            )
+            else -> ResponseEntity(500, createErrorBody(ApiError(ex.message.orEmpty(), "INTERNAL_SERVER_ERROR")))
         }
 
-    open fun <T> createResponse(contentType: MediaType?, input: APIGatewayProxyRequestEvent, response: ResponseEntity<T>): APIGatewayProxyResponseEvent {
-        return when {
-            // no-content response
-            response.body == null -> APIGatewayProxyResponseEvent()
+    open fun <T> createResponse(contentType: MediaType, response: ResponseEntity<T>): APIGatewayProxyResponseEvent =
+        when (response.body != null && serializationHandlerChain.supports(contentType, response.body)) {
+            true -> contentType
+            false -> MediaType.parse(router.defaultContentType)
+        }.let { finalContentType ->
+            APIGatewayProxyResponseEvent()
                 .withStatusCode(response.statusCode)
-                .withHeaders(response.headers)
-            serializationHandlerChain.supports(contentType!!, response.body) ->
-                APIGatewayProxyResponseEvent()
-                    .withStatusCode(response.statusCode)
-                    .withBody(serializationHandlerChain.serialize(contentType, response.body))
-                    .withHeaders(response.headers + ("Content-Type" to contentType.toString()))
-            else -> throw IllegalArgumentException("unsupported response ${response.body.let { (it as Any)::class.java }} and $contentType")
+                .withHeaders(response.headers + ("Content-Type" to finalContentType.toString()))
```

Comment:
Since we are using Guava anyway, we could make use of the constant `HttpHeaders.CONTENT_TYPE` here to avoid typos. What do you think?

Reply:
Yep, will do so! But to be honest, I would like to get rid of Guava (later) as it is about 2.6mb in size and is used just for parsing the Accept Header 🤣But using a constant is a good think anyway!

Reply:
Using our own constant would be fine for me, too :-)

Reply:
Also here 👍 for using constants - also let's get rid of the guava dependency and not add more usage here.